fluss-flink/fluss-flink-common/src/test/java/com/alibaba/fluss/flink/tiering/source/enumerator/TieringSourceEnumeratorTest.java

```diff
@@ -0,0 +1,635 @@
+                                genCommitLakeTableSnapshotRequest(
+                                        tableId,
+                                        partitionNameById.getKey(),
+                                        1,
+                                        partitionInitialBucketOffsets,
+                                        bucketOffsetOfInitialWrite.get(partitionNameById.getKey())))
+                        .get();
+            }
+            // notify this table tiering task finished
+            coordinatorGateway.lakeTieringHeartbeat(
+                    finishedTableHeartBeat(
+                            basicHeartBeat(),
+                            enumerator.getTieringTableEpochs(),
+                            enumerator.getFlussCoordinatorEpoch()));
+
+            Map<Long, Map<Integer, Long>> bucketOffsetOfSecondWrite =
+                    upsertRowForPartitionedTable(
+                            tablePath, DEFAULT_PK_TABLE_DESCRIPTOR, partitionNameByIds, 10, 20);
+            snapshotId = 1;
+            waitUntilPartitionTableSnapshot(tableId, partitionNameByIds, snapshotId);
+
+            // request tiering table splits
+            for (int subtaskId = 0; subtaskId < numSubtasks; subtaskId++) {
+                enumerator.handleSplitRequest(subtaskId, "localhost-" + subtaskId);
+            }
+
+            waitUntilTieringTableSplitAssignmentReady(
+                    context, DEFAULT_BUCKET_NUM * partitionNameByIds.size(), 500L);
+            List<TieringSplit> expectedLogAssignment = new ArrayList<>();
+            for (Map.Entry<Long, String> partitionNameById : partitionNameByIds.entrySet()) {
+                for (int tableBucket = 0; tableBucket < DEFAULT_BUCKET_NUM; tableBucket++) {
+                    long partionId = partitionNameById.getKey();
+                    expectedLogAssignment.add(
+                            new TieringLogSplit(
+                                    tablePath,
+                                    new TableBucket(tableId, partionId, tableBucket),
+                                    partitionNameById.getValue(),
+                                    bucketOffsetOfInitialWrite.get(partionId).get(tableBucket),
+                                    bucketOffsetOfInitialWrite.get(partionId).get(tableBucket)
+                                            + bucketOffsetOfSecondWrite
+                                                    .get(partionId)
+                                                    .get(tableBucket)));
+                }
+            }
+            List<TieringSplit> actualLogAssignment = new ArrayList<>();
+            for (SplitsAssignment<TieringSplit> splitsAssignment :
+                    context.getSplitsAssignmentSequence()) {
+                splitsAssignment.assignment().values().forEach(actualLogAssignment::addAll);
+            }
+            assertThat(actualLogAssignment).isEqualTo(expectedLogAssignment);
+        }
+        dropTable(tablePath);
```

Comment:
we don't need `dropTable` at the end of each test. 
`FlussClusterExtension` will clean all tables after each test method.